packages/zapp/console/src/components/Entities/EntityInputs.tsx

```diff
@@ -0,0 +1,226 @@
+import {
+  Paper,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  Typography,
+} from '@material-ui/core';
+import { makeStyles, Theme } from '@material-ui/core/styles';
+import CheckIcon from '@material-ui/icons/Check';
+import { useLaunchPlans } from 'components/hooks/useLaunchPlans';
+import { formatType, getInputDefintionForLiteralType } from 'components/Launch/LaunchForm/utils';
+import { FilterOperationName } from 'models/AdminEntity/types';
+import { ResourceIdentifier } from 'models/Common/types';
+import { LaunchPlanClosure, LaunchPlanSpec } from 'models/Launch/types';
+import * as React from 'react';
+import t from './strings';
+import { transformLiterals } from '../Literals/helpers';
+
+const useStyles = makeStyles((theme: Theme) => ({
+  header: {
+    marginBottom: theme.spacing(1),
+  },
+  divider: {
+    borderBottom: `1px solid ${theme.palette.divider}`,
+    marginBottom: theme.spacing(1),
+  },
+  rowContainer: {
+    display: 'flex',
+    marginTop: theme.spacing(3),
+  },
+  firstColumnContainer: {
+    width: '60%',
+    marginRight: theme.spacing(3),
+  },
+  secondColumnContainer: {
+    width: '40%',
+  },
+  configs: {
+    listStyleType: 'none',
+    paddingInlineStart: 0,
+  },
+  config: {
+    display: 'flex',
+  },
+  configName: {
+    color: theme.palette.grey[400],
+    fontSize: '14px',
+    marginRight: theme.spacing(2),
+    minWidth: '95px',
+  },
+  configValue: {
+    color: '#333',
+    fontSize: '14px',
+  },
+  headCell: {
+    fontSize: '14px',
+    color: theme.palette.grey[400],
+  },
+  noInputs: {
+    color: theme.palette.grey[400],
+  },
+}));
+
+interface Input {
+  name: string;
+  type?: string;
+  required?: boolean;
+  defaultValue?: string;
+}
+
+/** Fetches and renders the expected & fixed inputs for a given Entity (LaunchPlan) ID */
+export const EntityInputs: React.FC<{
+  id: ResourceIdentifier;
+}> = ({ id }) => {
+  const styles = useStyles();
+
+  const launchPlanState = useLaunchPlans(
+    { project: id.project, domain: id.domain },
+    {
+      limit: 1,
+      filter: [
+        {
+          key: 'launch_plan.name',
+          operation: FilterOperationName.EQ,
+          value: id.name,
+        },
+      ],
+    },
+  );
+
+  const closure = launchPlanState?.value?.length
+    ? launchPlanState.value[0].closure
+    : ({} as LaunchPlanClosure);
+
+  const spec = launchPlanState?.value?.length
+    ? launchPlanState.value[0].spec
+    : ({} as LaunchPlanSpec);
+
+  const expectedInputs = React.useMemo<Input[]>(() => {
+    const results = [] as Input[];
+    Object.keys(closure?.expectedInputs?.parameters || {}).forEach((name) => {
```

Comment:
It's better to use  ?? instead of || here , as

```
Object.keys(closure?.expectedInputs?.parameters ?? {}).forEach(...)
``` 
A ?? B   means use A if **defined** otherwise use B.
A || B     means use A if **not false** otherwise use B. which can lead to corner case problems for some datatypes where defined value could be interpreted as false. For example empty string or number 0 would be interpreted as false, but could be valid defined inputs. Such types of errors hard to fix in future.

Reply:
fixed